packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -191,12 +191,13 @@ exports.sourceNodes = async (
         setPluginStatus({ lastFetched: res.body.timestamp })
         requireFullRebuild = true
       } else {
-        // Touch nodes so they are not garbage collected by Gatsby.
-        getNodes().forEach(node => {
+        const nodes = getNodes()
+        for (let i = 0; i < nodes.length; i++) {
+          const node = nodes[i]
           if (node.internal.owner === `gatsby-source-drupal`) {
             touchNode(node)
           }
-        })
+        }
```

Comment:
This code path doesn't change anything right? we still loop over all nodes? I rather move it back to `forEach` as it's less code. Or did I miss something?

Reply:
These are micro-optimizations. `forEach` is maybe 2x slower than `for` loops — https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/

Maybe not worth doing 🤷 but node counts can get very large so these microoptimizations start to matter at scale esp for preview/inc builds.